vet-service/src/test/java/com/petclinic/vet/presentationlayer/VetControllerIntegrationTest.java

```diff
@@ -616,6 +617,99 @@ void getAverageRatingByVetId_withInvalidVetId_ShouldThrowNumberZero() {
+                .create(setup)
+                .expectNextCount(1)
+                .verifyComplete();
+
+        String existingDate = "2023";
+
+
+
+        RatingResponseDTO ratingWithDate = RatingResponseDTO.builder()
+                .date(rating1.getDate())
+                .ratingId(rating1.getRatingId())
+                .vetId(VET_ID)
+                .date(existingDate)
+                .rateScore(rating1.getRateScore())
+                .rateDescription(rating1.getRateDescription())
+                .predefinedDescription(rating1.getPredefinedDescription())
+                .build();
+
+        client
+                .get()
+                .uri("/vets/"+VET_ID+"/ratings/date?year={year}",existingDate)
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBodyList(RatingResponseDTO.class)
+                .value((list) -> {
+                    assertEquals(2, list.size());
+                    assertEquals(rating1.getRatingId(),list.get(0).getRatingId());
+                    assertEquals(rating1.getVetId(),list.get(0).getVetId());
+                    assertEquals(rating1.getRateScore(),list.get(0).getRateScore());
+                    assertEquals(rating1.getRateDate(),list.get(0).getRateDate());
+                    assertEquals(rating1.getRateDescription(),list.get(0).getRateDescription());
+                    assertEquals(existingDate, list.get(0).getDate());
+                });
+    }
+
+    @Test
+    void getRatingBasedOnYearDateWithInvalidVetID_ShouldFail() {
+
+        Publisher<Rating> setup = ratingRepository.deleteAll().
+                thenMany(ratingRepository.save(rating1));
+
+        StepVerifier
+                .create(setup)
+                .expectNextCount(1)
+                .verifyComplete();
+
+        String invalidVetId="123";
+        String existingDate = "2023";
+
+        client
+                .get()
+                .uri("/vets/"+invalidVetId+"/ratings/date?year={year}",existingDate)
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus().isNotFound()
+                .expectBody()
+                .jsonPath("$.message").isEqualTo("No valid ratings were found for "+invalidVetId);
+
+    }
+
+
+    @Test
+    void getRatingBasedOnYearDateWithInvalidYear_ShouldFail() {
+
+        Publisher<Rating> setup = ratingRepository.deleteAll().
+                thenMany(ratingRepository.save(rating1));
+
+        StepVerifier
+                .create(setup)
+                .expectNextCount(1)
+                .verifyComplete();
+
+        String invalidYear="203q";
+
+        client
+                .get()
+                .uri("/vets/"+VET_ID+"/ratings/date?year="+ invalidYear)
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus().isNotFound()
```

Comment:
In a later bug, this could be updated to ".expectStatus().isEqualTo(HttpStatus.UNPROCESSABLE_ENTITY)" but otherwise looks good.